### PR TITLE
prometheus backend: allow binding on addresses other than 127.0.0.1

### PR DESCRIPTION
**Pull request by deepfire**
issue: https://github.com/input-output-hk/iohk-monitoring-framework/issues/458

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [ ] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [ ] link to an epic
- [ ] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

File: iohk-monitoring/src/Cardano/BM/Data/Configuration.lhs [hunk 1]
```diff
@@ -42,6 +43,7 @@ import           Cardano.BM.Data.Rotation
 \subsubsection{Representation}\label{code:Representation}\index{Representation}\label{code:Port}\index{Port}
 \begin{code}
 type Port = Int
+type HostPort = (String, Int)
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Data/Configuration.lhs]: why not HostPort = (String, Port) ?

Comment (CodiePP) [reply iohk-monitoring/src/Cardano/BM/Data/Configuration.lhs]: or, do we still need "Port"

Comment (deepfire) [reply iohk-monitoring/src/Cardano/BM/Data/Configuration.lhs]: We still need `Port`, because it's used by other configuration

**Pull request merged**